### ISSUE TITLE
[UX] Add back the hide/show descriptions toggle in the Configuration page (/admin/config).

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#979]
Part of #506.

Possible improvements would be:
- [ ] make this work with JS magic (i.e. no extra page load)
- [ ] have the descriptions be hidden by default after a certain (TBD) breakpoint.

Here's how it's implemented in D8 and how it looks in narrow screens:

![d8_config_page-show_descriptions](https://cloud.githubusercontent.com/assets/2423362/8027295/b47ffb80-0dd8-11e5-8b50-d32093301015.png)

![d8_config_page-hide_descriptions](https://cloud.githubusercontent.com/assets/2423362/8027296/bc45acf2-0dd8-11e5-974f-6a5f14107850.png)

**Comment by klonos** [comment]
...actually this feature was not implemented in D8, so this is a regression from D7.

**Comment by markabur** [comment]
I agree, the show/hide descriptions control should come back. Not sure that they should necessarily be hidden on mobile though. The descriptions are either helpful to you or they're not anymore -- it doesn't have anything to do with what device you're on.

**Comment by Dinornis** [comment]
👍 on letting the user control the show/hide descriptions.

**Comment by jenlampton** [comment]
I'm surprised everyone here wants the show/hide button back. This is something that didn't do well in usability studies, which is why we removed it in Backdrop. I wonder if expectations are different when coming from Drupal vs having no or little Drupal experience?  If anyone who doesn't have Drupal experience comes across this issue and has an opinion, I'd love to hear it (just to be sure we've got everyone's feedback).

**Comment by olafgrabienski** [comment]
I *have* Drupal experience but don't want the *hide/show descriptions* back, I've tried the option once and then never used it again in Drupal.

And, in my opinion the descriptions shouldn't be hidden on narrow screens. We could however check the descriptions with the goal to shorten and improve them otherwise. This would improve readability both on desktop and on narrow screens.

**Comment by markabur** [comment]
As someone who uses the admin/config screen several times a day, I always hide the descriptions when possible, simply because it lets me work faster. Time is money!

What exactly didn't do well in usability testing? Having the descriptions hidden, or something about the show/hide control itself?

For me the descriptions actually impede usability, in a way I can feel -- like, I know from experience I should be able to hit "Performance" in 0.1 second after the page appears, but if I find myself hunting for it, it's probably because the pesky descriptions are cluttering up the screen. So, I take a moment to click Show/Hide so that I can find what I'm looking for faster the next time at least. 

The screenshots above are a great example. In the top one I can't see all the options at once, even on a big 27" screen. :-( It's cluttery and the text on the descriptions is uncomfortably small. The bottom one is beautiful: short lists, nicely grouped. Very easy to find a particular link quickly. I bet the bottom one would win in a race! (between experienced site builders, at least -- but maybe newbies too)

**Comment by alanmels** [comment]
The descriptions are good for new users, however if someone spent years developing on Drupal/Backdrop they become just unnecessary and, actually, annoying space takers. Anyway, we at AltaGrade have been using a custom module to hide descriptions on the configuration page. Didn't rolled it our, because it is very simple one, but let me know if someone needs it.
